internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,335 @@
+
+// getLogs gets the log-files list of a fuzz-target on a given day.
+func getLogs(date string, projectName string, fuzzTarget string, fuzzEngine string, sanitizer string) (*storage.BucketHandle, []string) {
+	// logsBucket is the ClusterFuzz Google Cloud Storage containing the fuzzers logs for a given project.
+	logsBucket := fmt.Sprintf("%s-logs.clusterfuzz-external.appspot.com", projectName)
+	bucket, err := getBucket(logsBucket)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fuzzengine := strings.ToLower(fuzzEngine)
+	// relativePath is the relative path in the logsBucket where the logs of a given fuzz-target on a given day are saved.
+	relativePath := fmt.Sprintf("%s_%s_%s/%s_%s_%s/%s", fuzzEngine, projectName, fuzzTarget, fuzzengine, sanitizer, projectName, date)
+	blobs, err := listBlobs(bucket, relativePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return bucket, blobs
+}
+
+// getFuzzEffortFromFile gets the fuzzingEffort from a single log file.
+func getFuzzEffortFromFile(reader io.Reader, revisionHash string, projectName string) (int, float64) {
+	var revisionHashInLog string
+	var err error
+	var timeFuzz float64
+	var numTests int
+	caser := cases.Title(language.English)
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		// get the revisionHash mentioned in the logfile.
+		if strings.Contains(scanner.Text(), caser.String(projectName)) {
+			revisionHashInLog = strings.Split(scanner.Text(), " ")[1]
+		}
+		// get the fuzzing time in seconds
+		if strings.Contains(scanner.Text(), "Time ran:") {
+			timeFuzz, err = strconv.ParseFloat(strings.Split(scanner.Text(), " ")[2], 32)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		// get the number of fuzz-tests
+		if strings.Contains(scanner.Text(), "stat::number_of_executed_units") {
+			numTests, err = strconv.Atoi(strings.Split(scanner.Text(), " ")[1])
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	// return the fuzzing efforts if the revisionHash in the logfile
+	// is the same as the revisionHash we are considering (it is
+	// possible to have logs for multiple revisions a given day)
+	if revisionHashInLog == revisionHash {
+		return numTests, timeFuzz
+	}
+	// the fuzzing effort does not count if the revisionHash in the
+	// logfile is different from the revisionHash we are considering.
+	return 0, 0.0
+}
+
+// crashDetected detects crashes in log files that are related to a given revisionHash
+// and a given day.
+func crashDetected(reader io.Reader, revisionHash string, projectName string) bool {
+	var revisionHashInLog string
+	scanner := bufio.NewScanner(reader)
+	caser := cases.Title(language.English)
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), caser.String(projectName)) {
+			revisionHashInLog = strings.Split(scanner.Text(), " ")[1]
+		}
+		if strings.Contains(scanner.Text(), "fuzzer-testcases/crash-") {
+			if revisionHashInLog == revisionHash {
+				return true
+			}
+		}
+	}
```

Comment:
Is not an error check needed here? Similar to 

```
	if err := scanner.Err(); err != nil {
		...
	}
```

Reply:
I updated the crash detection to optimise it since we don't need to search line by line to find the crash.  So, I'm not using the scanner in the newer version.

Reply:
I replaced the `scanner` in `crashDetected` to optimise the detection of crashes since I don't need to scan the logs line by line to find the crashes.